lib/date_range.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "reservation_booker"
+require 'date'
+
+module Hotel
+    class DateRange
+      def self.valid_checkin_dates?(check_in_year, check_in_month, check_in_day)
```

Comment:
For the whole of the `DateRange` class, it would have been okay to pass in a `Date` object instead of passing in the day, month and year. You can rely on `Date` to make sure that the date is properly formatted.